warehouse/accounts/forms.py

```diff
@@ -32,6 +33,18 @@ def validate_username(self, field):
             raise wtforms.validators.ValidationError("No user found with that username")
 
 
+class TOTPValueMixin:
+
+    totp_value = wtforms.StringField(
+        validators=[
+            wtforms.validators.DataRequired(),
+            wtforms.validators.Regexp(
+                rf"^[0-9]{{{TOTP_LENGTH}}}$", message="TOTP code must be 6 digits."
```

Comment:
This is a nitpick (especially since this length is unlikely to ever change), but the message should probably be `message=f"TOTP code must be {TOTP_LENGTH} digits."`